lib/builder/items.rb

```diff
@@ -2,33 +2,34 @@
 
 module ItemBuilder
   class Item
-    attr_accessor :items,
+    attr_accessor :item_details,
                   :id,
                   :name,
                   :price,
                   :quantity
 
     def build
-      copy_from_individual_fields unless items_exist?
-      items.map! do |part|
-        item = ::Item.new
-        item.id = part[:id]
-        item.name = part[:name]
-        item.price = part[:price]
-        item.quantity = part[:quantity]
-        item
-      end
-      items
+      copy_from_individual_fields unless item_exist?
+      item = ::Item.new
+      item.id = item_details[:id]
+      item.name = item_details[:name]
+      item.price = item_details[:price]
+      item.quantity = item_details[:quantity]
+      item
+    end
+
+    def add_item(item)
+      @item_details = item
```

Comment:
maybe it would be more readable if we renamed item => item_details so as not to get confused with when item is being used for the Item domain object?

Reply:
Yeah I like that, so use item_details when it's a hash and item when it's a domain object?

Reply:
could this method take named parameters?